src/classes/ACCT_CannotDelete_TEST.cls

```diff
@@ -36,9 +36,834 @@
+        newContacts[1].Primary_Organization__c = newAccounts[1].Id; 
+        insert newContacts; 
+        
+        Test.startTest();
+        Database.DeleteResult[] results = Database.delete(newAccounts, false);
+        Test.stopTest();
+        
+        List<Account> returnAccounts = [SELECT Id
+                                        FROM Account
+                                        WHERE Id IN :newAccounts];
+        
+        System.assertEquals(2, returnAccounts.size());
+        System.assertEquals(Label.CannotDelete, results[0].errors[0].message);
+    }
+    
+    /*********************************************************************************************************
+    * @description Test method to test if Prevent_Account_Deletion__c is enabled in Hierarchy Settings, and
+    * Account has a Course Connection record associated to it, that it cannot be deleted.
+    */
+    @isTest 
+    public static void accCannotDeleteWithCourseConnection() {
+        UTIL_CustomSettingsFacade.getSettingsForTests(new Hierarchy_Settings__c
+                                                         (Account_Processor__c = bizAccRecordTypeId,
+                                                          Prevent_Account_Deletion__c = TRUE));
+        
+        List<Account> newAccounts = UTIL_UnitTestData_TEST.getMultipleTestAccounts(2, bizAccRecordTypeId); 
+        insert newAccounts; 
+        
+        List<Contact> newContacts = UTIL_UnitTestData_TEST.getMultipleTestContacts(2); 
+        newContacts[0].LastName = 'Willy';
+        newContacts[0].AccountId = newAccounts[0].Id; 
+        newContacts[1].LastName = 'Jones'; 
+        newContacts[1].AccountId = newAccounts[1].Id; 
+        insert newContacts; 
+        
+        List<Course__c> courses = new List<Course__c>{
+            new Course__c (Name = 'Course1', Account__c  = newAccounts[0].Id),
+            new Course__c (Name = 'Course2', Account__c  = newAccounts[1].Id)
+        };        
+        insert courses; 
+        
+        List<Term__c> terms = new List<Term__c>{
+            new Term__c(Name = 'Fall', Account__c = newAccounts[0].Id),
+            new Term__c(Name = 'Spring', Account__c = newAccounts[1].Id)  
+        }; 
+        insert terms; 
+        
+        Course_Offering__c courseOffering1 = UTIL_UnitTestData_TEST.createCourseOffering(courses[0].Id, terms[0].Id); 
+        Course_Offering__c courseOffering2 = UTIL_UnitTestData_TEST.createCourseOffering(courses[1].Id, terms[1].Id); 
+        
+        Course_Enrollment__c courseConnection1 = UTIL_UnitTestData_TEST.getCourseConnection(newContacts[0].Id, courseOffering1.Id); 
+        Course_Enrollment__c courseConnection2 = UTIL_UnitTestData_TEST.getCourseConnection(newContacts[1].Id, courseOffering2.Id); 
+        courseConnection1.Account__c = newAccounts[0].Id;
+        courseConnection2.Account__c = newAccounts[1].Id;
+        insert new SObject[] {courseConnection1, courseConnection2}; 
```

Comment:
We seem to be using this insert new SObject[]{} format in some places which differs from the simple insert list format that we have in other pieces of code. Is there a reason we aren't using the same format?

Reply:
@MatthewBlanski , no good reason :( I'll update it to be consistent. 👍